dune/stuff/la/solver/eigen.hh

```diff
@@ -337,21 +340,21 @@ public:
     // check for inf or nan
     const bool check_for_inf_nan = opts.get("check_for_inf_nan", default_opts.get< bool >("check_for_inf_nan"));
     if (check_for_inf_nan) {
-      // serialize matrix (no copy done here)
-      MatrixType& non_const_ref = const_cast< MatrixType& >(matrix_);
-      const EigenMappedDenseVector< S > values(non_const_ref.backend().valuePtr(), non_const_ref.backend().nonZeros());
-      for (size_t ii = 0; ii < values.size(); ++ii) {
-        const S& val = values[ii];
-        if (std::isnan(val) || std::isinf(val))
-          DUNE_THROW(Exceptions::linear_solver_failed_bc_data_did_not_fulfill_requirements,
-                     "Given matrix contains inf or nan and you requested checking (see options below)!\n"
-                     << "If you want to disable this check, set 'check_for_inf_nan = 0' in the options.\n\n"
-                     << "Those were the given options:\n\n"
-                     << opts);
+      //iterates over the non-zero entries of matrix_.backend() and checks them
+      typedef typename MatrixType::BackendType::InnerIterator InnerIterator;
+      for (int ii = 0; ii < matrix_.backend().outerSize(); ++ii) {
```

Comment:
Could you please add a private
```c++
typedef typename BackendType::Index EIGEN_size_t;
```
(as in https://github.com/wwu-numerik/dune-stuff/blob/master/dune/stuff/la/container/eigen/sparse.hh#L89), and use `EIGEN_size_t` instead of `int`?